cloud-arm/src/main/java/com/sequenceiq/cloudbreak/cloud/arm/ArmUtils.java

```diff
@@ -171,4 +177,68 @@ public String getCustomSubnetId(Network network) {
         return network.getStringParameter(SUBNET_ID);
     }
 
+    public void validateSubnetRules(AzureRMClient client, Network network) {
+        if (isExistingNetwork(network)) {
+            String resourceGroupName = getCustomResourceGroupName(network);
+            String networkId = getCustomNetworkId(network);
+            String subnetId = getCustomSubnetId(network);
+            Map subnetProperties = client.getSubnetProperties(resourceGroupName, networkId, subnetId);
+            Map networkSecurityGroup = (Map) subnetProperties.get("networkSecurityGroup");
+            if (networkSecurityGroup != null) {
+                validateSecurityGroup(client, networkSecurityGroup);
+            }
+        }
+    }
+
+    private void validateSecurityGroup(AzureRMClient client, Map networkSecurityGroup) {
+        String securityGroupId = (String) networkSecurityGroup.get("id");
+        String[] parts = securityGroupId.split("/");
+        if (parts.length != ID_SEGMENTS) {
+            LOGGER.info("Cannot get the security group's properties, id: {}", securityGroupId);
+            return;
+        }
+        Map securityGroupProperties = client.getSecurityGroupProperties(parts[RG_PART], parts[SEC_GROUP_PART]);
+        LOGGER.info("Retrieved security group properties: {}", securityGroupProperties);
+        List securityRules = (List) securityGroupProperties.get("securityRules");
+        boolean port22Found = false;
+        boolean port443Found = false;
+        for (Object securityRule : securityRules) {
+            Map rule = (Map) securityRule;
+            Map properties = (Map) rule.get("properties");
+            if (isValidInboundRule(properties)) {
+                String destinationPortRange = (String) properties.get("destinationPortRange");
+                if ("*".equals(destinationPortRange)) {
+                    return;
+                }
+                String[] range = destinationPortRange.split("-");
+                port443Found = port443Found ? port443Found : isPortFound(PORT_443, range);
+                port22Found = port22Found ? port22Found : isPortFound(PORT_22, range);
+                if (port22Found && port443Found) {
+                    return;
+                }
+            }
+        }
+        throw new CloudConnectorException("The specified subnet's security group does not allow traffic for port 22 and/or 443");
+    }
+
+    private boolean isValidInboundRule(Map properties) {
+        Object protocol = properties.get("protocol");
+        return "Inbound".equals(properties.get("direction")) && ("Tcp".equals(protocol) || "*".equals(protocol)) && "Allow".equals(properties.get("access"));
```

Comment:
"inbound".equals(properties.get("direction").tolowercase()) maybe much better because then we not depending on azure api changes